cfgmgr/teammgr.cpp

```diff
@@ -373,7 +417,25 @@ bool TeamMgr::addLagMember(const string &lag, const string &member)
     // ip link set dev <member> down;
     // teamdctl <port_channel_name> port add <member>;
     cmd << IP_CMD << " link set dev " << member << " down; ";
-    cmd << TEAMDCTL_CMD << " " << lag << " port add " << member << "; ";
+    cmd << TEAMDCTL_CMD << " " << lag << " port add " << member;
+
+    if (exec(cmd.str(), res) != 0)
+    {
+        // teamdctl port add command will fail when the member port is not
+        // set to admin status down; it is possibel that some other processes
+        // or users (e.g. portmgrd) are executing the command to bring up the
+        // member port while adding this port into the port channel. This piece
+        // of code will check if the port is set to admin status up. If yes,
+        // it will retry to add the port into the port channel.
+        if (checkPortIffUp(member))
+        {
+            return task_need_retry;
+        }
+        else
+        {
+            return task_failed;
```

Comment:
>task_failed [](start = 19, length = 11)

in case of task_failed, I think we should print error message as it mean a port member is not able to be added to port channel.